ibis/backends/clickhouse/compiler/core.py

```diff
@@ -45,14 +45,37 @@
 
 import sqlglot as sg
 
+import ibis.common.patterns as pats
+import ibis.expr.analysis as an
 import ibis.expr.operations as ops
 import ibis.expr.types as ir
 from ibis.backends.clickhouse.compiler.relations import translate_rel
+from ibis.backends.clickhouse.compiler.values import translate_val
 
 if TYPE_CHECKING:
     from collections.abc import Mapping
 
 
+p = pats.Object.namespace(ops)
+
+
+def cumulative_to_window(
+    func: ops.Cumulative, frame: ops.WindowFrame
+) -> ops.WindowFunction:
+    klass = getattr(ops, func.__class__.__name__.replace("Cumulative", ""))
+    new_op = klass(*func.args)
+    new_frame = frame.copy(start=None, end=0)
+    new_expr = an.windowize_function(new_op.to_expr(), frame=new_frame)
+    return new_expr.op()
+
+
+def _translate_node(node, *args, **kwargs):
```

Comment:
I can do that in a follow up.

Reply:
Actually I think this adds unnecessary implicit behavior:

1. I now have to deal with import cycles
2. I now have to ensure that both `values` and `relations` `translate` is imported so that all the rules are registered.

I don't really see what we gain by making these into a single rule.

Reply:
Put all the translate implementations into a single file?

Reply:
I like that they are separate now, it makes it easier to things IMO.